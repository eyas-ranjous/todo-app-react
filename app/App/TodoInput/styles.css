.todo-input {
  margin-bottom: 10px;
  font-size: 1em;
  width: 100%;
}

.todo-input input {
  padding: 10px 15px;
  border: 1px solid #e5e3e3;
  outline: none;
  border-radius: 20px;
}
