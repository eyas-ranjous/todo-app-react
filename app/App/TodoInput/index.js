import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styles from './styles.css';

const TodoInput = () => {
  const dispatch = useDispatch();
  let todoInput;

  const addTodo = () => {
    if (!todoInput.value) {
      todoInput.classList.add(styles.required);
      return;
    }
    todoInput.classList.remove(styles.required);
    dispatch({
      type: 'ADD_TODO',
      todo: { text: todoInput.value }
    });
    todoInput.value = '';
  };

  const handleKeyDown = (e) => {
    if (e.keyCode === 13) {
      addTodo();
    }
  };

  const TextInput = () => (
    <input
      type="text"
      className={styles['todo-text-input']}
      onKeyDown={handleKeyDown}
      placeholder="Write Todo Text..."
      ref={(el) => { todoInput = el; }}
    />
  );

  const AddButton = () => (
    <button onClick={() => addTodo()}>Add</button>
  );

  return (
    <div className={styles['todo-input']}>
      <TextInput />
      <AddButton />
    </div>
  );
};

export default TodoInput;
