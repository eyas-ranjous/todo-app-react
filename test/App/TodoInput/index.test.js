import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux'
import createStore from '../../../app/store';
import TodoInput from '../../../app/App/TodoInput';

describe('TodoInput', () => {
  const store = createStore();

  it('renders successfully', () => {
    const wrapper = mount(
      <Provider store={store}>
        <TodoInput />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
