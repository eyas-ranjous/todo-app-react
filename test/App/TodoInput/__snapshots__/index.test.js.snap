// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput renders successfully 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <TodoInput>
    <div
      className="todo-input"
    >
      <input
        className="todo-text-input"
        onKeyDown={[Function]}
        placeholder="Write Todo Text..."
        type="text"
      />
      <button
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </TodoInput>
</Provider>
`;
